shell/common/pointer_data_dispatcher.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef POINTER_DATA_DISPATCHER_H_
+#define POINTER_DATA_DISPATCHER_H_
+
+#include "flutter/runtime/runtime_controller.h"
+#include "flutter/shell/common/animator.h"
+
+namespace flutter {
+
+class PointerDataDispatcher {
+ public:
+  virtual void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                              uint64_t trace_flow_id) = 0;
+  virtual void OnRender() = 0;
+  virtual ~PointerDataDispatcher() {}
+};
+
+class DefaultPointerDataDispatcher : public PointerDataDispatcher {
+ public:
+  DefaultPointerDataDispatcher(Animator& animator,
+                               RuntimeController& runtime_controller)
+      : runtime_controller_(runtime_controller), animator_(animator) {}
+
+  void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
```

Comment:
Though overridden methods don't need to be documented (they are documented in the base class), we add a comment above each overridden method listing the class defining the virtual method. See [this file for the pattern](https://github.com/flutter/engine/blob/a353f93d370dded7bc7756082fbab8b519679d4d/shell/common/shell.h#L462). Please add `// |PointerDataDispatcer|` above this method for example.